### PR TITLE
builder: add amazon-2023 target

### PR DESCRIPTION
**Pull request by Habbie**
### Checklist

I have:
- [ ] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: builder-support/specs/pdns.spec [hunk 1]
```diff
@@ -141,7 +144,7 @@ Summary: Geo backend for %{name}
 Group: System Environment/Daemons
 Requires: %{name}%{?_isa} = %{version}-%{release}
 BuildRequires: yaml-cpp-devel
-%if 0%{?rhel} < 9
+%if 0%{?rhel} < 9 && 0%{?amzn} != 2023
```
Comment (dwfreed) [inline builder-support/specs/pdns.spec:147]: If RHEL older than 9 (nine), or Amazon not 2023, or (neither RHEL nor Amazon)

File: builder-support/specs/pdns.spec [hunk 2]
```diff
@@ -203,7 +206,9 @@ export LDFLAGS=-L/usr/lib64/boost169
   --with-lua=%{lua_implementation} \
   --with-dynmodules='%{backends}' \
   --enable-tools \
+%if 0%{?amzn} != 2023
```
Comment (dwfreed) [inline builder-support/specs/pdns.spec:209]: As above, If Amazon not 2023 or not Amazon

**Comment by Habbie** [comment]
@dwfreed thanks for checking my logic!

File: builder-support/specs/dnsdist.spec [hunk 3]
```diff
@@ -112,8 +114,9 @@ make %{?_smp_mflags} check || (cat test-suite.log && false)
 install -d %{buildroot}/%{_sysconfdir}/dnsdist
 %{__mv} %{buildroot}%{_sysconfdir}/dnsdist/dnsdist.conf-dist %{buildroot}%{_sysconfdir}/dnsdist/dnsdist.conf
 chmod 0640 %{buildroot}/%{_sysconfdir}/dnsdist/dnsdist.conf
-sed -i "s,/^\(ExecStart.*\)dnsdist\(.*\)\$,\1dnsdist -u dnsdist -g dnsdist\2," %{buildroot}/lib/systemd/system/dnsdist.service
-sed -i "s,/^\(ExecStart.*\)dnsdist\(.*\)\$,\1dnsdist -u dnsdist -g dnsdist\2," %{buildroot}/lib/systemd/system/dnsdist@.service
+find / -name dnsdist.service
```
Comment (rgacogne) [inline builder-support/specs/dnsdist.spec:117]: Do we want to keep the `find`?

**Review by rgacogne — APPROVED** [review]

**Review by rgacogne — APPROVED** [review]

**Pull request merged**